prusti-interface/src/environment/polonius_info.rs

```diff
@@ -1029,25 +1065,56 @@ impl<'a, 'tcx: 'a> PoloniusInfo<'a, 'tcx> {
 
     /// Convert a facts::Loan to LoanPlaces<'tcx> (if possible)
     pub fn get_loan_places(&self, loan: &facts::Loan) -> Option<LoanPlaces<'tcx>> {
-        let loan_location = self.loan_position[loan];
-        let mir_block = &self.mir[loan_location.block];
-        if loan_location.statement_index < mir_block.statements.len() {
-            let mir_stmt = &mir_block.statements[loan_location.statement_index];
-            match mir_stmt.kind {
-                mir::StatementKind::Assign(ref lhs_place, ref rvalue) => Some(LoanPlaces {
-                    dest: lhs_place.clone(),
-                    source: rvalue.clone(),
-                    location: loan_location,
-                }),
-
-                ref x => {
-                    debug!("Borrow starts at statement {:?}", x);
-                    None
-                }
+        let location = self.loan_position[loan];
+        let assignment = self.get_assignment_for_loan(*loan)?;
+        let (dest, source) = assignment.as_assign().unwrap();
+        let dest = dest.clone();
+        let source = source.clone();
+        Some(LoanPlaces { dest, source, location })
+    }
```

Comment:
Implementing `get_loan_places` is a bit more complicated when there are tuples. This is because an assignment like

    _3 = (move _4, move _5)

creates two loans on the same line (if both fields of `_3` are references), let's call them `L0` and `L1`. This means we can't just inspect the assign statement to figure out the loan places for `L0` -- we also have to find out which field of `_3` `L0` is associated with. This happens in `get_assignment_for_loan`, which returns an atomic assignment for a given loan. For the example above, `get_assignment_for_loan(L0)` would be `_3.0 = move _4` and `get_assignment_for_loan(L1)` would be `_3.1 = move _5`. Using these atomic assignments, we can simply read off the loan places as before.